spokestack/extensions/webrtc/agc.pyx

```diff
@@ -0,0 +1,94 @@
+"""
+This module contains the AutomaticGainControl class implemented in Cython
+"""
+# distutils: sources = filter_audio/agc/analog_agc.c, filter_audio/agc/digital_agc.c
+# distutils: include_dirs = filter_audio/agc/include/
+cimport numpy as np
+cimport cagc
+
+
+MIC_MAX = 255
+MIC_TARGET = 180
+
+np.import_array()
+
+cdef class AutomaticGainControl:
+    cdef void*_agc
+    cdef int _sample_rate
+    cdef int _frame_width
+    cdef int _target_level_dbfs
+    cdef int _compression_gain_db
+    cdef int _limit_enable
+    cdef cagc.WebRtcAgc_config_t _config
+
+    def __dealloc__(self):
+        cagc.WebRtcAgc_Free(self._agc)
+
+    def __init__(self,
+                 sample_rate=16000,
+                 frame_width=20,
+                 target_level_dbfs=0,
+                 compression_gain_db=0,
+                 limit_enable=True,
+                 **kwargs):
+
+        self._sample_rate = sample_rate
+        self._frame_width = frame_width
+        self._target_level_dbfs = target_level_dbfs
+        self._compression_gain_db = compression_gain_db
+        self._limit_enable = limit_enable
+
+        self._agc = NULL
+        result = cagc.WebRtcAgc_Create(&self._agc)
+        if result < 0:
+            raise ValueError
+        if result == 0:
+            result = cagc.WebRtcAgc_Init(self._agc,
+                                         minLevel=0,
+                                         maxLevel=MIC_MAX,
+                                         agcMode=2,
+                                         fs=self._sample_rate)
+
+            if result == 0:
+                self._config.targetLevelDbfs = self._target_level_dbfs
+                self._config.limiterEnable = self._limit_enable
+                self._config.compressionGaindB = self._compression_gain_db
+                result = cagc.WebRtcAgc_set_config(self._agc, self._config)
+        if result < 0:
```

Comment:
same here, want to check `result != 0`, and I would add a symbolic message to that `ValueError` (`invalid_config` or whatever)